src/js/containers/award/shared/ContractGrantActivityContainer.jsx

```diff
@@ -42,42 +43,61 @@ const ContractGrantActivityContainer = ({ awardId, awardType, dates }) => {
      */
     const formatTransactions = (rawTransactions) => {
         // Reduce into unique transaction objects
-        let newData = rawTransactions.reduce((acc, data) => {
-            const updatedData = { ...data };
-            updatedData.action_date = moment(updatedData.action_date, 'YYYY-MM-DD');
-            const currentTransactionIndex = acc.findIndex((x) => x.action_date.valueOf() === updatedData.action_date.valueOf());
-            /**
-             * When we have multiple transactions on the same day, we will sum their obligation and
-             * we will keep track of all the transactions on the same date for the tooltips in the
-             * allTransactions property.
-             */
-            if (currentTransactionIndex !== -1) {
-                // update the allTransactions array if it exists
-                if (acc[currentTransactionIndex]?.allTransactions) {
-                    acc[currentTransactionIndex].allTransactions.push(updatedData);
-                }
-                else {
+        let newData = rawTransactions
+            .sort((a, b) => a.action_date.valueOf() - b.action_date.valueOf())
+            .reduce((acc, data) => {
+                const updatedData = { ...data };
+                updatedData.action_date = moment(updatedData.action_date, 'YYYY-MM-DD');
+                const currentTransactionIndex = acc.findIndex((x) => x.action_date.valueOf() === updatedData.action_date.valueOf());
+                /**
+                 * When we have multiple transactions on the same day, we will sum their obligation and
+                 * we will keep track of all the transactions on the same date for the tooltips in the
+                 * allTransactions property.
+                 */
+                if (currentTransactionIndex !== -1) {
+                    // update the allTransactions array if it exists
+                    if (acc[currentTransactionIndex]?.allTransactions) {
+                        acc[currentTransactionIndex].allTransactions.push(updatedData);
+                    }
+                    else {
+                        /**
+                         * The first duplicate found.
+                         * We will add the duplicate which is this data, and we will add the
+                         * original node which is acc[currentTransactionIndex].
+                         */
+                        const clonedTransaction = cloneDeep(acc[currentTransactionIndex]);
+                        acc[currentTransactionIndex].allTransactions = [clonedTransaction, updatedData];
```

Comment:
What if there is more than one duplicate transaction? We would be overwriting the `allTransactions` property to only have two; right? Also, clonedTransaction might have a `allTransactions` property. Right?

If this is right, maybe we need to do:

```suggestion
                        acc[currentTransactionIndex].allTransactions = Object.keys(clonedTransaction).includes('allTransactions')
                          ? [...acc[currentTransactionIndex].allTransactions, data]
                          :  [acc[currentTransactionIndex], data];
```